### PR TITLE
Add BF16 IE Precision specialization to is_float

### PR DESCRIPTION
**Pull request by apertovs**
Ticket: 55748

### CODE REVIEW

**Comment by apertovs** [comment]
@ilyachur, have a look, please.

File: inference-engine/include/ie_precision.hpp [hunk 1]
```diff
@@ -430,18 +430,20 @@ inline uint8_t type_size_or_zero<void>() {
     return 0;
 }
 
+namespace {
+constexpr bool isSpecificFloatPrecision(const Precision::ePrecision& precision) {
```
Comment (ilyachur) [inline inference-engine/include/ie_precision.hpp:434]: Could you add new tests?

Comment (apertovs) [reply inference-engine/include/ie_precision.hpp:434]: Added

**Comment by apertovs** [comment]
@ilyachur, have a look, please.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**